indexer/services/ender/src/validators/validator.ts

```diff
@@ -23,6 +26,16 @@ export abstract class Validator<T> {
    */
   public abstract validate(): void;
 
+  /**
+   * Returns the decoded event and any additional details the SQL function needs to process.
+   */
+  // TODO(IND-513): Convert handlers to have a 1-1 relationship with each validator by merging
+  // the order fill handlers together into a single handler (and the respective SQL function)
+  // and push this method down into the handler itself.
+  public getEventForBlockProcessor(): Promise<object> {
```

Comment:
Is there a way to return a typed object here?

Reply:
It is `T` + some additional fields (really only the order cancellation status for the order fill handler uses this currently). Is there a good way to represent that?

Reply:
we could just use `T` for now and add a comment that OrderFill id different in the OrderFillValidator and in validator.ts. I don't think that would cause an error, but if does then we can leave it like this and file a bug ticket

Reply:
Could you have a second type in the generic declaration? Like:
```
export abstract class Validator<T, V extends T>  {
...
getExtendedEvent(event: T): V {
  return event;
}
...
public getEventForBlockProcessor(): V {
  return getExtendedevent(Promise.resolve(this.event));
}
```

and then in the order fill handler you have V be the type with the extra fields and override `getExtendedEvent`?

I don't think this is too important though, the above approach would add a lot of extra type declarations to every other handler that doesn't need it.

If we don't add it, perhaps just add a comment that this is `object` because of the logic in order fill handler?

Reply:
I got:
```TS2322: Type  T  is not assignable to type  V 
 T  is assignable to the constraint of type  V , but  V  could be instantiated with a different subtype of constraint  object```
 
 Swapped to use `T` since that is technically true since it is a sub-type of `T`.